python/CMakeLists.txt

```diff
@@ -49,10 +49,20 @@ set_target_properties(pyarb PROPERTIES OUTPUT_NAME _arbor)
 #   arbor.cpython-36m-x86_64-linux-gnu.so
 set_target_properties(pyarb PROPERTIES PREFIX "${PYTHON_MODULE_PREFIX}" SUFFIX "${PYTHON_MODULE_EXTENSION}")
 
+# Set RPATH for the installation phase
+if(APPLE)
```

Comment:
I am fine with that solution, although I was originally thinking about using
it in header-only mode still. Nothing that can't be change later, though, if
needed.